### PR TITLE
Update messaging_interactions, add two new endpoints

### PR DESCRIPTION
**Pull request by tomekzbrozek**
# Description of change

- [x] update `messaging_interactions` endpoint by upgrading to API v2 and starting to explicitly list [contentToRetrieve](https://developers.liveperson.com/messaging-interactions-api-methods-conversations.html#note-new-capability---partial-retrieval-of-data) - in this case, there are two new columns added above default, i.e. `"unAuthSdes"` and `"monitoring"`. This allows to see sell conversion.
- [x] add new endpoint: `agent_state_distribution`: https://developers.liveperson.com/operational-realtime-api-methods-agent-activity.html
- [x] add new endpoint: `messaging_queue_health`: https://developers.liveperson.com/messaging-operations-api-methods-messaging-queue-health.html

# Manual QA steps
 - example JSON outputs have been sense-checked as well as data loaded to Snowflake via `target-stitch`
 
# Risks
 - two new columns are added to `messaging_interactions` table which may cause backward compatibility issues
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Comment by cmerrick** [comment]
Hi @tomekzbrozek, thanks for your contribution!

In order for us to evaluate and accept your PR, we ask that you [sign a contribution license agreement](https://stitch-cla-enforcer.herokuapp.com/). It's all electronic and will take just minutes.

**Comment by cmerrick** [comment]
You did it @tomekzbrozek!

Thank you for signing the Singer Contribution License Agreement.

File: tap_liveperson/streams/messaging_queue_health.py
Comment (luandy64) [inline tap_liveperson/streams/messaging_queue_health.py]: Does `metricsData` and `skillsMetrics` always get returned here?

I think it would be safer to do `records.get('metricsData', {}).get('skillsMetrics', {}).items()`

File: tap_liveperson/streams/messaging_queue_health.py
Comment (luandy64) [inline tap_liveperson/streams/messaging_queue_health.py:29]: Does `records` always have a `timestamp`?

File: tap_liveperson/schemas/messaging_queue_health.json
Comment (luandy64) [inline tap_liveperson/schemas/messaging_queue_health.json]: Can we make all of these `"type"` nullable? Or do you have docs that say these are always returned?

File: setup.py
Comment (luandy64) [inline setup.py:6]: For future reference, leave this for the maintainers to do

File: tap_liveperson/streams/base.py
Comment (luandy64) [inline tap_liveperson/streams/base.py]: I'm not a fan of using exceptions as control flow.

Could this be something like
```
if self.CONTENT_TO_RETRIEVE:
  return {"contentToRetrieve": self.CONTENT_TO_RETRIEVE}
return {}
```

File: tap_liveperson/schemas/messaging_queue_health.json
Comment (luandy64) [reply tap_liveperson/schemas/messaging_queue_health.json]: For example, instead of:
```
"skill": {
    "type": "string"
},
```
You could have:
```
"skill": {
    "type": ["string",  "null"]
},
```

**Review by luandy64 — CHANGES_REQUESTED** [review]

File: tap_liveperson/streams/messaging_queue_health.py
Comment (tomekzbrozek) [reply tap_liveperson/streams/messaging_queue_health.py]: yes it should return metrics on a skill level (so `skillsMetrics`), because we pass `"skillIds": "all",` to request parameters, but I agree that we can be on the safe side and do it as you suggested!

File: tap_liveperson/streams/messaging_queue_health.py
Comment (tomekzbrozek) [reply tap_liveperson/streams/messaging_queue_health.py:29]: yes, this particular endpoint returns some metrics aggregated by a timestamp that is determined by `interval` parameter passed to request parameters. So it's always going to be present in the results.

Documentation for this endpoint: https://developers.liveperson.com/messaging-operations-api-methods-messaging-queue-health.html

File: tap_liveperson/schemas/messaging_queue_health.json
Comment (tomekzbrozek) [reply tap_liveperson/schemas/messaging_queue_health.json]: 👍

**Comment by tomekzbrozek** [comment]
@luandy64 all suggestions have been addressed and changes applied, thanks! 
This one is ready for a re-review!

**Review by luandy64 — APPROVED** [review]

**Comment by tomekzbrozek** [comment]
Thanks @luandy64 and one last question: any idea how this tap will handle two new columns that will appear in `messaging_interactions` stream once it is merged? Will they be skipped for previous implementations they assumed lack of these columns, will they be added and backfilled with NULL values before date of deployment to production, or something else...? Thanks!

PS. I'm referring to `monitoring` and `unAuthSdes`: https://github.com/singer-io/tap-liveperson/pull/1/files#diff-52715fa86fffa5b9ba15ff02b22cfe29R545-R605

**Comment by tomekzbrozek** [comment]
@jacobrobertbaca thanks for merging, would you mind easing my conscience and referring to my last question (above)? 😁 🙏

**Comment by jacobrobertbaca** [comment]
Hi @tomekzbrozek , if by previous implementation you mean previous syncs, these columns will just not be included in all syncs prior to this merge, but will be included in all syncs following. It's similar to what happens when a stream is not selected on one sync but selected on the subsequent sync. Does that answer your question?

**Comment by tomekzbrozek** [comment]
@jacobrobertbaca but if the target is a single table in a database (e.g. in Snowflake), will this table contain the two new columns after syncs following this PR, or will they be skipped instead?

**Comment by jacobrobertbaca** [comment]
For all of the syncs after the merge, if you select the new columns, then you should see them in Snowflake for data created after your bookmark date. And for data in there right now, I would expect some kind of null value for the new columns. If you want all the data (including that before the update) for the `messaging_interactions` stream, you can reset your bookmark to the start date.

**Comment by tomekzbrozek** [comment]
this makes sense, thanks a lot, @jacobrobertbaca !

**Pull request merged**
